apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts

```diff
@@ -153,48 +126,74 @@ export class SendMessagePush extends SendMessageBase {
       return;
     }
 
-    if (integration) {
-      for (const channel of pushChannels) {
-        if (!channel.credentials?.deviceTokens) {
-          await this.createExecutionDetails.execute(
-            CreateExecutionDetailsCommand.create({
-              ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
-              detail: DetailEnum.PUSH_MISSING_DEVICE_TOKENS,
-              source: ExecutionDetailsSourceEnum.INTERNAL,
-              status: ExecutionDetailsStatusEnum.FAILED,
-              isTest: false,
-              isRetry: false,
-            })
-          );
-
-          continue;
-        }
-
-        await this.sendMessage(
-          integration,
-          channel.credentials.deviceTokens,
-          title,
-          content,
-          command,
-          notification,
-          command.payload,
-          overrides,
-          channel.providerId
+    let sent = false;
+    for (const channel of pushChannels) {
+      if (!channel.credentials?.deviceTokens) {
+        await this.createExecutionDetails.execute(
+          CreateExecutionDetailsCommand.create({
+            ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
+            detail: DetailEnum.PUSH_MISSING_DEVICE_TOKENS,
+            source: ExecutionDetailsSourceEnum.INTERNAL,
+            status: ExecutionDetailsStatusEnum.FAILED,
+            isTest: false,
+            isRetry: false,
+          })
+        );
+
+        continue;
+      }
+
+      const integration = await this.getIntegration(
+        GetDecryptedIntegrationsCommand.create({
+          organizationId: command.organizationId,
+          environmentId: command.environmentId,
+          channelType: ChannelTypeEnum.PUSH,
+          providerId: channel.providerId,
+          findOne: true,
+          active: true,
+          userId: command.userId,
+        })
+      );
+
+      if (!integration) {
+        await this.createExecutionDetails.execute(
+          CreateExecutionDetailsCommand.create({
+            ...CreateExecutionDetailsCommand.getDetailsFromJob(command.job),
+            detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION,
+            source: ExecutionDetailsSourceEnum.INTERNAL,
+            status: ExecutionDetailsStatusEnum.FAILED,
+            isTest: false,
+            isRetry: false,
+          })
         );
+
+        continue;
       }
 
+      const overrides = command.overrides[integration.providerId] || {};
+
+      sent = true;
+      await this.sendMessage(
+        integration,
+        channel.credentials.deviceTokens,
+        title,
+        content,
+        command,
+        notification,
+        command.payload,
+        overrides,
+        channel.providerId
+      );
+    }
+
+    if (sent) {
       return;
     }
 
-    await this.sendErrors(pushChannels, command, notification, integration);
+    await this.sendErrors(pushChannels, command, notification);
```

Comment:
This is a good catch.

Reply:
But I noticed that the functionality implemented here is the same as it is happening in lines 114-124.
I'd remove from line 115 to 124 and replace for this call `await this.sendErrors` and early return.
This will help us to remove the `flag` you added for the loop in line 129, that I find it a bit useless.

Reply:
Done, just need to push